app/src/main/java/com/github/dodobest/upbitapi/CoinListPageFragment.kt

```diff
@@ -0,0 +1,85 @@
+package com.github.dodobest.upbitapi
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.DividerItemDecoration
+import com.github.dodobest.upbitapi.databinding.FragmentCoinListPageBinding
+import com.github.dodobest.upbitapi.model.MarketPlaceName
+import dagger.hilt.android.AndroidEntryPoint
+import timber.log.Timber
+
+@AndroidEntryPoint
+class CoinListPageFragment : Fragment() {
+
+    private lateinit var marketPlaceName: MarketPlaceName
+    private lateinit var upbitAdapter: UpbitAdapter
+
+    private var _binding: FragmentCoinListPageBinding? = null
+    private val binding get() = _binding ?: throw IllegalArgumentException(getString(R.string.view_binding_is_null))
+
+    private val viewModel: UpbitViewModel by viewModels()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        arguments?.takeIf { it.containsKey(Constant.ARGUMENT_MARKET_INDEX) }?.apply {
+            MarketPlaceName.from(getInt(Constant.ARGUMENT_MARKET_INDEX))?.let {
+                marketPlaceName = it
+            } ?: throw IllegalArgumentException(getString(R.string.no_exist_market))
+        } ?: throw IllegalArgumentException(getString(R.string.no_exist_market))
+        setRecyclerView()
+        setLiveDataObserve()
+        loadInitialContent()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentCoinListPageBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    private fun loadInitialContent() {
+        viewModel.getMarkets(marketPlaceName.toString())
+    }
+
+    private fun setLiveDataObserve() {
+        viewModel.tickers.observe(viewLifecycleOwner) {
+            upbitAdapter.setResult(it)
+        }
+    }
+
+    private fun setRecyclerView() {
+        upbitAdapter = UpbitAdapter(marketPlaceName)
+        binding.coinPriceRecyclerView.adapter = upbitAdapter
+
+        context?.let { context ->
+            DividerItemDecoration(
+                context,
+                DividerItemDecoration.VERTICAL
+            ).let { dividerItemDecoration ->
+                AppCompatResources.getDrawable(context, R.drawable.divider)?.let {
+                    dividerItemDecoration.setDrawable(it)
+                } ?: run {
+                    Timber.e(getString(R.string.error_create_divider_drawable))
+                    return
+                }
+
+                binding.coinPriceRecyclerView.addItemDecoration(dividerItemDecoration)
+            }
+        } ?: throw IllegalArgumentException(getString(R.string.no_exist_context))
```

Comment:
Context는 if문으로 null 처리가 되지 않아 let을 사용했습니다.

```
// Smart cast to 'Context' is impossible, because 'context' is a property that has open or custom getter
if (context != null) {
    AppCompatResources.getDrawable(context, R.drawable.divider) !=  null
}
```

DividerItemDecoration과 getDrawable에 let을 사용한 이유는 한 번만 사용되는 값을 변수로 저장하고 싶지 않아서 입니다!
viewBinding을 변수로 가지고 있는 것 만으로도 메모리 leak을 유발할 수 있다는 사실을 알고, 코드를 이렇게 구현해봤습니다!

Reply:
[88b63c5](https://github.com/DoTheBestMayB/UpbitAPI/pull/35/commits/88b63c594a45394b451cb7e7643edc7a7615620d)

Reply:
> DividerItemDecoration과 getDrawable에 let을 사용한 이유는 한 번만 사용되는 값을 변수로 저장하고 싶지 않아서입니다.

중첩 if문이 가독성이 좋지 않은 것과 마찬가지로, 중첩 범위 연산자 또한 가독성이 좋지 않습니다.
변수는 재활용할 수 있는 경우에도 사용하지만, 가독성을 위해서 사용하는 경우도 있습니다.

Reply:
1 depth 원칙을 준수하도록 노력하겠습니다!